analytics/terraform/spark-k8s-operator/raydata-pipeline/main.tf

```diff
@@ -0,0 +1,313 @@
+    ]
+    resources = [
+      "arn:aws:glue:${var.aws_region}:${local.account_id}:catalog",
+      "arn:aws:glue:${var.aws_region}:${local.account_id}:database/*",
+      "arn:aws:glue:${var.aws_region}:${local.account_id}:table/*/*"
+    ]
+  }
+
+  # Additional permissions for Iceberg operations
+  statement {
+    effect = "Allow"
+    actions = [
+      "lakeformation:GetDataAccess"
+    ]
+    resources = ["*"]
+  }
+}
+
+# Create IAM policy with prefixed name
+resource "aws_iam_policy" "s3_access" {
+  name        = local.iam_policy_name
+  description = "S3 and Glue access policy for Ray Spark log processing with Iceberg"
+  policy      = data.aws_iam_policy_document.s3_access.json
+  tags        = local.common_tags
+}
+
+# =============================================================================
+# IAM Role for IRSA (IAM Roles for Service Accounts)
+# =============================================================================
+
+# Trust policy for IRSA
+data "aws_iam_policy_document" "irsa_trust" {
+  statement {
+    effect = "Allow"
+    principals {
+      type = "Federated"
+      identifiers = [
+        "arn:aws:iam::${local.account_id}:oidc-provider/oidc.eks.${var.aws_region}.amazonaws.com/id/${local.oidc_issuer_id}"
+      ]
+    }
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+    condition {
+      test     = "StringEquals"
+      variable = "oidc.eks.${var.aws_region}.amazonaws.com/id/${local.oidc_issuer_id}:sub"
+      values   = ["system:serviceaccount:${var.namespace}:${var.service_account_name}"]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "oidc.eks.${var.aws_region}.amazonaws.com/id/${local.oidc_issuer_id}:aud"
+      values   = ["sts.amazonaws.com"]
+    }
+  }
+}
+
+# Create IAM role with prefixed name
+resource "aws_iam_role" "ray_s3_access" {
+  name               = local.iam_role_name
+  description        = "IAM role for Ray Spark log processing with Iceberg and IRSA"
+  assume_role_policy = data.aws_iam_policy_document.irsa_trust.json
+  tags               = local.common_tags
+}
+
+# Attach policy to role
+resource "aws_iam_role_policy_attachment" "ray_s3_access" {
+  role       = aws_iam_role.ray_s3_access.name
+  policy_arn = aws_iam_policy.s3_access.arn
+}
+
+# =============================================================================
+# AWS Glue Database for Iceberg
+# =============================================================================
+
+resource "aws_glue_catalog_database" "raydata_iceberg" {
+  name        = var.iceberg_database
+  description = "Database for storing processed Spark logs using Iceberg format"
+
+  catalog_id = local.account_id
+
+  tags = local.common_tags
+}
+
+# =============================================================================
+# Kubernetes Resources using kubectl provider
+# =============================================================================
+
+# Create namespace
+resource "kubectl_manifest" "namespace" {
```

Comment:
lets use this file https://github.com/awslabs/data-on-eks/blob/main/analytics/terraform/spark-k8s-operator/spark-team.tf to add another team called `raydata-team`

Reply:
so that we dont need to create namespace, service account, iam role etc in this file